### DISCUSSION TITLE
lazydev cannot highlight `vim` global variable

### DISCUSSION THREAD

**Discussion by Jaehaks** [folke/lazydev.nvim#42] [Q&A] [answered]
when using lazydev, it doesn't highlight `vim` variable as neodev did
How can I modify?

**Comment by folke** [comment, marked as answer]
That was removed in Neovim / treesitter.
You can use these queries to bring that back: https://github.com/LazyVim/LazyVim/tree/main/queries/lua

**Reply by Jaehaks** [reply]
Thank you for your reply :)

**Comment by DrKJeff16** [comment]
You can also attempt to manually configure your LSP to detect it, e.g. for`lua-language-server` you can set the config field
`Lua.diagnostics.globals = ["vim"]`.

Example using [neoconf](https://github.com/folke/neoconf.nvim):

```jsonc
{
    "lspconfig": {
        "lua_ls": {
            "Lua.diagnostics.globals" = [
                "vim",
                // ...
            ],
            // ...
        },
        // ...
    },
    // ...
}
```

Hope this helsps as it help me!

**Reply by folke** [reply]
op is refering to highlighting the `vim` variable. That's treesitter, not globals. 
With lazydev you should defintely **not** configure any globals.

**Comment by rafaelpirolla** [comment]
```
      {
        "folke/lazydev.nvim",
        ft = "lua", -- only load on lua files
        opts = {
          library = {
            -- See the configuration section for more details
            -- Load luvit types when the `vim.uv` word is found
            { path = "${3rd}/luv/library", words = { "vim%.uv", "vim%.loop" } },
            { path = "snacks.nvim", words = { "Snacks" } },
            { path = "lazy.nvim", words = { "LazyVim" } },
            { path = "~/.config/hammerspoon/Spoons/EmmyLua.spoon/annotations", words = { "hs" } },
          },
        },
      },
```
Old topic but how to get around without configuring a global for "hs", though – the above config still produces Undefined global 'hs'.
